src/libs/HttpUtils.js

```diff
@@ -98,7 +99,16 @@ function xhr(command, data, type = CONST.NETWORK.METHOD.POST, shouldUseSecure =
 
     let apiRoot = shouldUseSecure ? CONFIG.EXPENSIFY.SECURE_EXPENSIFY_URL : CONFIG.EXPENSIFY.URL_API_ROOT;
 
-    if (CONFIG.IS_IN_STAGING && shouldUseStagingServer) {
+    // getEnvironment promise returns the environment based on ENVIRONMENT Config for web/desktop.
+    // Additionally for ios/android betaChecker.isBetaBuild() does the version check
+    // environment is initialized as null as the default set in CONFIG.js always takes precedence
+    let environment = null;
+    Environment.getEnvironment()
+        .then((env) => {
+            environment = env;
+        });
+
+    if (environment !== CONST.ENVIRONMENT.PRODUCTION && shouldUseStagingServer) {
```

Comment:
@mountiny in line 12 of this file, the shouldUseStagingServer is default set to TRUE [callback: val => shouldUseStagingServer = lodashGet(val, 'shouldUseStagingServer', true)]. Which means in a DEV environment localhost shall request for staging api_url by default based on this condition (if (environment !== CONST.ENVIRONMENT.PRODUCTION && shouldUseStagingServer) ) returning TRUE.

Which results in CORS error. To avoid that it is better when shouldUseStagingServer is default set to FALSE. Can we do that ?

Reply:
@smrutiparida This would mean the Staging app will by default point to production and we dont want that.

Contributors can use the app agains staging API and they should, internal developer use dev agains local machine.

Reply:
>Contributors can use the app agains staging API and they should, internal developer use dev agains local machine.
Changed accordingly